### DISCUSSION TITLE
W5500 - SKR2

### DISCUSSION THREAD

**Discussion by boboxx** [grblHAL/core#415] [General]
I'm trying to add ethernet to my SKR2 via a W5500 module (https://www.aliexpress.com/item/4000226162371.html). I built my FW with the Web builder and enable the "Plugin_networking"

I have connected like this using the bellow maping but i'm unable to get a IP and I don't see any request from DHCP.
W5500 - SKR2 [Pin assignement]
CS - IO15 (ESP-CS) [PIN:PB12,SPI CS]
RST - RST (ESP-RST) [PIN:PB11,SPI reset]
SCK - IO14 (ESP IO14) [PIN:PB13,SCK]
M - IO12 (ESP-MISO) [PIN:PB14,MISO]
MO - IO13 (ESP-MOSI )[PIN:PB15,MOSI]

$i
[VER:1.1f.20231210:]
[OPT:VNMSL,100,1024,3,0]
[AXS:3:XYZ]
[NEWOPT:ENUMS,RT+,ES,SED,ETH,SD]
[FIRMWARE:grblHAL]
[NVS STORAGE:*FLASH]
[FREE MEMORY:26K]
[DRIVER:STM32F407]
[DRIVER VERSION:231202]
[BOARD:BTT SKR-2]
[WIZCHIP:W5500]
[IP:]
[PLUGIN:MODBUS v0.15]
[PLUGIN:HUANYANG VFD v0.11]
[SPINDLE:PWM]
[PLUGIN:SDCARD v1.11]
[PLUGIN:FS macro plugin v0.06]

$Pins
[PIN:PA4,Emergency stop]
[PIN:PA5,Feed hold]
[PIN:PA6,Cycle start]
[PIN:PE4,Probe]
[PIN:PC1,X limit min]
[PIN:PC3,Y limit min]
[PIN:PC0,Z limit min]
[PIN:PB10,SPI IRQ]
[PIN:PE2,X step]
[PIN:PD5,Y step]
[PIN:PA15,Z step]
[PIN:PE1,X dir]
[PIN:PD4,Y dir]
[PIN:PA8,Z dir]
[PIN:PC13,Stepper power]
[PIN:PE3,X enable]
[PIN:PD6,Y enable]
[PIN:PD1,Z enable]
[PIN:PB6,Spindle on]
[PIN:PB5,Spindle direction]
[PIN:PB3,Flood]
[PIN:PB4,Mist]
[PIN:PB12,SPI CS]
[PIN:PB11,SPI reset]
[PIN:PA10,RX,Modbus]
[PIN:PA9,TX,Modbus]
[PIN:PD9,RX,UART2]
[PIN:PD8,TX,UART2]
[PIN:PB0,Spindle PWM]
[PIN:PB13,SCK]
[PIN:PB14,MISO]
[PIN:PB15,MOSI]

**Comment by terjeio** [comment]
`[PIN:PB10,SPI IRQ]` is not connected to the INT pin?

**Reply by boboxx** [reply]
Ok added SPI IRQ, same behaviour. I see the green power and yellow activity blinking but no IP listed in controler.

W5500 - SKR2 [Pin assignement]
IN - IO0 (ESP-IO0) [PIN:PB10,SPI IRQ]
CS - IO15 (ESP-CS) [PIN:PB12,SPI CS]
RST - IO4 (ESP-IO4) [PIN:PB11,SPI reset]
SCK - IO14 (ESP IO14) [PIN:PB13,SCK]
M - IO12 (ESP-MISO) [PIN:PB14,MISO]
MO - IO13 (ESP-MOSI )[PIN:PB15,MOSI]

**Reply by boboxx** [reply]
These are the pin identification I'm using:

![pinout](https://github.com/grblHAL/core/assets/15091037/358248de-97db-4d7f-872d-16875bc3b4b2)
![W5500](https://github.com/grblHAL/core/assets/15091037/9192fcc9-7f43-434b-b07b-5de736379593)

**Reply by terjeio** [reply]
There is a resistor in series with the interrupt pin that I have overlooked, I'll commit a fix soon - likely tomorrow.

**Reply by terjeio** [reply]
Fix committed, new pin mappings [here](https://github.com/grblHAL/STM32F4xx/blob/95706f4fa97fec80fd02d79b0c9b8aa7aeb1b041/Inc/btt_skr_2.0_map.h#L184-L187).
I have not tested with a SKR board as I do not have any, code works ok with a Nucleo development board.

**Reply by boboxx** [reply]
Still not able to get it working with the new pins:
[PIN:PB11,SPI IRQ]
[PIN:PB12,SPI CS]
[PIN:PB13,SCK]
[PIN:PB14,MISO]
[PIN:PB15,MOSI]
[PIN:PC14,SPI reset]


Also the version is still reporting old version:
[VER:1.1f.20231222:]
[OPT:VNMSL,100,1024,3,0]
[AXS:3:XYZ]
[NEWOPT:ENUMS,RT+,ES,SED,ETH]
[FIRMWARE:grblHAL]
[NVS STORAGE:*FLASH]
[FREE MEMORY:29K]
[DRIVER:STM32F407]
[DRIVER VERSION:231223]
[BOARD:BTT SKR-2]
[WIZCHIP:W5500]
[MAC:56:34:12:dc:00:00]
[IP:]
[PLUGIN:MODBUS v0.15]
[PLUGIN:HUANYANG VFD v0.11]
[SPINDLE:PWM]

**Reply by terjeio** [reply]
Try with not connecting the reset pin.
I have assigned the pins according to the schematic, I have no means to verify that it is correct.
Do you have access to an oscilloscope or logic analyszer?

> Also the version is still reporting old version:

This is correct, the version number is for the core and that is functionally the same.

**Reply by boboxx** [reply]
yay that worked, thank you!!!


[PIN:PC14,SPI reset] //ESP-RST
[PIN:PB11,SPI IRQ] //ESP-IO4
[PIN:PB12,SPI CS] //ESP-CS
[PIN:PB13,SCK] //ESP-CLK
[PIN:PB14,MISO] //ESP-MISO
[PIN:PB15,MOSI] //ESP-MOSI



GrblHAL 1.1f ['$' or '$HELP' for help]
[VER:1.1f.20231222:]
[OPT:VNMSL,100,1024,3,0]
[AXS:3:XYZ]
[NEWOPT:ENUMS,RT+,ES,SED,ETH]
[FIRMWARE:grblHAL]
[NVS STORAGE:*FLASH]
[FREE MEMORY:29K]
[DRIVER:STM32F407]
[DRIVER VERSION:231223]
[BOARD:BTT SKR-2]
[WIZCHIP:W5500]
[MAC:56:34:12:dc:00:00]
[IP:192.168.11.40]
[NETCON:Websocket]
[PLUGIN:MODBUS v0.15]
[PLUGIN:HUANYANG VFD v0.11]
[SPINDLE:PWM]
